### PR TITLE
Move balance monitor

### PR DESCRIPTION
**Pull request by cedric-cordenier**

### CODE REVIEW

File: common/txmgr/types/config.go [hunk 1]
```diff
@@ -10,30 +10,37 @@ type TxmConfig[FEE_UNIT Unit] interface {
 	ReaperConfig
 
 	SequenceAutoSync() bool
-	UseForwarders() bool
-	MaxQueuedTransactions() uint64
+}
+
+type TxmTransactionsConfig interface {
```
Comment (jmank88) [inline common/txmgr/types/config.go:15]: Maybe s/Transaction/Tx all around? :thinking: 
```suggestion
type TxmTxsConfig interface {
```

Comment (cedric-cordenier) [reply common/txmgr/types/config.go:15]: I would rather keep it as Transactions as all of the other interfaces use the same pattern (using the antecedent accessor in the config var name).

Also my brain finds it hard to parse TxmTxs :P

Comment (jmank88) [reply common/txmgr/types/config.go:15]: But then why not this?
```suggestion
type TransactionManagerTransactionsConfig interface {
```

File: core/chains/evm/config/config.go [hunk 2]
```diff
@@ -80,6 +73,26 @@ type ChainScopedOnlyConfig interface {
 
 	// OCR2 chain specific config
 	OCR2AutomationGasLimit() uint32
+
+	EVM() EVM
+}
+
+type EVM interface {
```
Comment (jmank88) [inline core/chains/evm/config/config.go:80]: This seems odd - do we need this extra indirection? IIRC `ChainScopedOnlyConfig` is already only EVM, without any global/root config to distinguish from.

Comment (jmank88) [reply core/chains/evm/config/config.go:80]: Oh could we have this method on `ChainScopedConfig` instead? That seems like a cleaner separation :thinking:

Comment (cedric-cordenier) [reply core/chains/evm/config/config.go:80]: Yeah this was the bit I wasn't super sure about. My reason for moving everything EVM-related to cfg.EVM() was because that's the config key everything is under.

I can move it to ChainScopedConfig or remove it entirely if you'd rather go that way? (I'd probably prefer to move it since I do think there's some sense to having the EVM keys under `cfg.EVM()`, but don't have super strong feelings either way)

File: common/txmgr/types/config.go (hunk 1 above)
Comment (cedric-cordenier) [reply common/txmgr/types/config.go]: Ha, good point! I've addressed

File: core/chains/evm/config/config.go (hunk 2 above)
Comment (cedric-cordenier) [reply core/chains/evm/config/config.go]: (I've moved to ChainScopedConfig but lmk if you'd rather I removed entirely)

Comment (jmank88) [reply core/chains/evm/config/config.go]: Sounds good. Yeah I see that `.Evm()` makes sense at the callsite with the composed interface :+1:

File: core/chains/evm/config/config.go [hunk 3]
```diff
@@ -82,9 +75,29 @@ type ChainScopedOnlyConfig interface {
...
+	ResendAfterThreshold() time.Duration
+	ReaperThreshold() time.Duration
+	MaxInFlight() uint32
+	MaxQueued() uint64
+}
+
 //go:generate mockery --quiet --name ChainScopedConfig --output ./mocks/ --case=underscore
 type ChainScopedConfig interface {
 	config.AppConfig
 	ChainScopedOnlyConfig
```
Comment (jmank88) [inline core/chains/evm/config/config.go:99]: So then the goal would be to dop this whole thing, right? Maybe leave a note, and/or deprecate?
```suggestion
	ChainScopedOnlyConfig // Deprecated
```
IIRC the linter will even object to using deprecated things.

**Review by george-dorin — APPROVED** [review]
